ci/staging-acceptance-tests.sh

```diff
@@ -0,0 +1,213 @@
+if [[ "$api_target" != *"fr-stage"* ]]; then
+  echo "### THIS IS A DESTRUCTIVE TEST to the sandbox-fedramp org, DO NOT RUN IN PRODUCTION ###"
+  echo "Error: Not targeting staging. Current API endpoint: $api_target, exiting for your own safety."
+  exit 1
+fi
+
+# Create a random 32-character password with hyphens
+PASSWORD=$(cat /dev/urandom | base64 | tr -dc '0-9a-zA-Z' | head -c32)
+
+# Create the user
+echo "Creating CF user test.user@fedramp.gov with a random password..."
+cf create-user "test.user@fedramp.gov" "$PASSWORD" >/dev/null 2>&1
+
+# Sleep for 45 seconds, app runs every 30 seconds
+echo "Sleeping test for 45 seconds to wait for sandbox-bot app to process the new user..."
+sleep 45
+
+# Observe the output from the app, it should create a new org, space and quotas within 30 seconds
+ORG_NAME="sandbox-fedramp"
+SPACE_NAME="test.user"
+EXPECTED_ORG_QUOTA="sandbox-fedramp"
+EXPECTED_SPACE_QUOTA="sandbox_quota"
+REQUIRED_SECURITY_GROUPS=("public_networks_egress" "trusted_local_networks_egress")
+
+echo "🔍 Checking for org '$ORG_NAME'..."
+if cf orgs | grep -q "^$ORG_NAME$"; then
+  echo "✅ Org '$ORG_NAME' found. Targeting..."
+  cf target -o "$ORG_NAME" >/dev/null 2>&1
+else
+  echo "❌ Org '$ORG_NAME' not found."
+fi
+
+echo "🔍 Checking org quota for '$ORG_NAME'..."
+org_quota=$(cf org "$ORG_NAME" | awk -F': ' '/quota:/ {print $2}' | xargs || echo "UNKNOWN")
+if [[ "$org_quota" == "$EXPECTED_ORG_QUOTA" ]]; then
+  echo "✅ Org quota is '$org_quota'"
+else
+  echo "❌ Org quota mismatch. Expected '$EXPECTED_ORG_QUOTA', got '$org_quota'"
+fi
+
+echo "🔍 Checking spaces for org '$ORG_NAME'..."
+spaces_output=$(cf spaces)
+space_count=$(echo "$spaces_output" | grep -c "^$SPACE_NAME$")
+if [[ "$space_count" -eq 1 ]]; then
+  echo "✅ Space '$SPACE_NAME' found."
+  cf target -s "$SPACE_NAME" >/dev/null 2>&1
+else
+  echo "❌ Expected one space named '$SPACE_NAME', found $space_count"
+fi
+
+echo "🔍 Checking space quota for space '$SPACE_NAME'..."
+space_quota=$(cf space "$SPACE_NAME" | awk -F': ' '/quota:/ {print $2}' | xargs || echo "UNKNOWN")
+if [[ "$space_quota" == "$EXPECTED_SPACE_QUOTA" ]]; then
+  echo "✅ Space quota is '$space_quota'"
+else
+  echo "❌ Space quota mismatch. Expected '$EXPECTED_SPACE_QUOTA', got '$space_quota'"
+fi
+
+echo "🔍 Verifying running security groups from 'cf space $SPACE_NAME' output..."
+space_output=$(cf space "$SPACE_NAME")
+
+for group in "${REQUIRED_SECURITY_GROUPS[@]}"; do
+  if grep -A 10 "running security groups" <<< "$space_output" | grep -q "$group"; then
+    echo "✅ Running security group '$group' is listed in space config"
+  else
+    echo "❌ Missing required running security group '$group' in space config"
+  fi
+done
+
+echo "🔍 Checking org quota resource limits from 'cf org-quota $ORG_NAME'..."
+org_quota_output=$(cf org-quota "$ORG_NAME")
+
+# Total memory
+total_memory=$(awk -F': ' '/total memory:/ {print $2}' <<< "$org_quota_output" | xargs)
```

Comment:
If `xargs` is for trimming whitespace, I somewhat prefer `tr -d '[:space:]'` as being more intuitive

Reply:
Same comment goes for the rest of PR for trimming whitespace with `xargs`